decompiler/structures/pseudo/z3_logic.py

```diff
@@ -94,6 +108,8 @@ def _get_operation(self, operation: Operation) -> Union[BoolRef, BitVecRef]:
         operands = self._ensure_same_sort([self.convert(operand) for operand in operation.operands])
         if isinstance(operands[0], BoolRef) and operation.operation in self.OPERATIONS_BOOLREF:
             converter = self.OPERATIONS_BOOLREF.get(operation.operation, None)
+        elif all(is_bool(op) for op in operands) and operation.operation in self.OPERATIONS_INVALID_BOOLREF_OP:
```

Comment:
Why not `operation.operation not in self.OPERATIONS_BOOLREF`? These are the only operations where we can do something, or do I miss something? Otherwise, a few are missing.

Second Question: Do you need the `all(is_boll(op))` or is it enough to check the first one?  I thought the `ensure_same_sort` ensures that all operands have the same type. But not sure.

Reply:
Second question first:
You are right. Changed in next commit.

First question:
I only wanted to override the operations which will generate a error. 
If a operation is not in `OPERATIONS_INVALID_BOOLREF_OP` the converter will be used from `OPERATIONS` in the line after that. 

If I would use your condition I would need to implement every possible lambda in `OPERATIONS_INVALID_BOOLREF_OP`.